test/e2e_kubeadm/dns_addon_test.go

```diff
@@ -0,0 +1,150 @@
+)
+
+var (
+	dnsType = ""
+)
+
+// Define container for all the test specification aimed at verifying
+// that kubeadm configures the dns as expected
+var _ = KubeadmDescribe("dns addon", func() {
+
+	// Get an instance of the k8s test framework
+	f := framework.NewDefaultFramework("dns")
+
+	// Tests in this container are not expected to create new objects in the cluster
+	// so we are disabling the creation of a namespace in order to get a faster execution
+	f.SkipNamespaceCreation = true
+
+	// kubeadm supports two type of DNS addon, and so
+	// it is necessary to get it from the kubeadm-config ConfigMap before testing
+	BeforeEach(func() {
+		// if the dnsType name is already known exit
+		if dnsType != "" {
+			return
+		}
+
+		// gets the ClusterConfiguration from the kubeadm kubeadm-config ConfigMap as a untyped map
+		m := GetClusterConfiguration(f.ClientSet)
+
+		// Extract the dnsType
+		dnsType = "CoreDNS"
+		if _, ok := m["dns"]; ok {
+			d := m["dns"].(map[interface{}]interface{})
+			if t, ok := d["type"]; ok {
+				dnsType = t.(string)
+			}
+		}
+	})
+
+	Context("kube-dns", func() {
+		Context("kube-dns ServiceAccount", func() {
+			It("should exist", func() {
+				if dnsType != "kube-dns" {
+					framework.Skipf("Skipping because proxy type is %s", dnsType)
+				}
+
+				ExpectServiceAccount(f.ClientSet, kubeSystemNamespace, kubeDNSServiceAccountName)
+			})
+		})
+
+		Context("kube-dns Deployment", func() {
+			It("should exist and be properly configured", func() {
+				if dnsType != "kube-dns" {
+					framework.Skipf("Skipping because proxy type is %s", dnsType)
+				}
+
+				d := GetDeployment(f.ClientSet, kubeSystemNamespace, kubeDNSDeploymentName)
+
+				Expect(d.Spec.Template.Spec.ServiceAccountName).To(Equal(kubeDNSServiceAccountName))
+			})
+		})
+	})
+
+	Context("coreDNS", func() {
+		Context("coreDNS ServiceAccount", func() {
+			It("should exist", func() {
+				if dnsType != "CoreDNS" {
+					framework.Skipf("Skipping because proxy type is %s", dnsType)
+				}
+
+				ExpectServiceAccount(f.ClientSet, kubeSystemNamespace, coreDNSServiceAccountName)
+			})
+
+			It("should have related ClusterRole and ClusterRoleBinding", func() {
+				if dnsType != "CoreDNS" {
+					framework.Skipf("Skipping because proxy type is %s", dnsType)
+				}
+
+				ExpectClusterRole(f.ClientSet, coreDNSRoleName)
+				ExpectClusterRoleBinding(f.ClientSet, coreDNSRoleBindingName)
+			})
+		})
+
+		Context("coreDNS ConfigMap", func() {
+			It("should exist and be properly configured", func() {
+				if dnsType != "CoreDNS" {
+					framework.Skipf("Skipping because proxy type is %s", dnsType)
+				}
+
+				cm := GetConfigMap(f.ClientSet, kubeSystemNamespace, coreDNSConfigMap)
+
+				Expect(cm.Data).To(HaveKey(coreDNSConfigMapKey))
+			})
+		})
+
+		Context("coreDNS Deployment", func() {
+			It("should exist and be properly configured", func() {
+				if dnsType != "CoreDNS" {
+					framework.Skipf("Skipping because proxy type is %s", dnsType)
```

Comment:
proxy type - > DNS type
should we add a wrapper / helper that has the message only one time and does the framework skip?
e.g. `maybeSkipDNSType(framework, dnsType, "CoreDNS")`